presto-main/src/main/java/com/facebook/presto/resourcemanager/ResourceManagerClusterStateProvider.java

```diff
@@ -141,8 +144,10 @@ public void registerQueryHeartbeat(String nodeId, BasicQueryInfo basicQueryInfo)
     {
         requireNonNull(nodeId, "nodeId is null");
         requireNonNull(basicQueryInfo, "basicQueryInfo is null");
+        Stream<InternalNode> activeOrShuttingDownCoordinators = concat(internalNodeManager.getCoordinators().stream(),
+                internalNodeManager.getNodes(SHUTTING_DOWN).stream().filter(InternalNode::isCoordinator));
```

Comment:
it might make sense to expose Shutting down coordinator as a seperate method from InternalNodeManager. i.e. 'getShuttingDownCoordinator` or something similar.

Reply:
Also wondering, what value are we going to get with this check here where a coordinator sends it's heartbeat to the RM and we check if it belongs to the same set of nodes or not? We don't do same for the node heartbeat. @tdcmeehan  Any specific reason for this check here or can we remove this entirely?

Reply:
I don't see any other reason other than a scenario where RM receives a heart beat from a coordinator that does not belong to the cluster, not sure if this is possible. I will wait for Tim to comment.

Reply:
It's more of a check to ensure that we're not making decisions based on heartbeats which are inconsistent with discovery service.  I would like to eventually add a similar check for node heartbeats.